hibernate-core/src/main/java/org/hibernate/id/PostInsertIdentityPersister.java

```diff
@@ -7,41 +7,14 @@
 package org.hibernate.id;
 
 import org.hibernate.persister.entity.EntityPersister;
-import org.hibernate.persister.entity.mutation.EntityMutationTarget;
 
 /**
  * A persister that may have an identity assigned by execution of 
  * a SQL {@code INSERT}.
  *
  * @author Gavin King
  */
-public interface PostInsertIdentityPersister extends EntityPersister, EntityMutationTarget {
-	/**
-	 * Get a SQL select string that performs a select based on a unique
-	 * key determined by the given property name.
-	 *
-	 * @param propertyName The name of the property which maps to the
-	 *           column(s) to use in the select statement restriction.
-	 * @return The SQL select string
-	 */
-	String getSelectByUniqueKeyString(String propertyName);
-
-	/**
-	 * Get a SQL select string that performs a select based on a unique
-	 * key determined by the given property names.
-	 *
-	 * @param propertyNames The names of the properties which maps to the
-	 *               column(s) to use in the select statement restriction.
-	 * @return The SQL select string
-	 */
-	default String getSelectByUniqueKeyString(String[] propertyNames) {
-		// default impl only for backward compatibility
-		if ( propertyNames.length > 1 ) {
-			throw new IllegalArgumentException( "support for multiple properties not implemented" );
-		}
-		return getSelectByUniqueKeyString( propertyNames[0] );
-	}
-
+public interface PostInsertIdentityPersister extends EntityPersister {
```

Comment:
I'm wondering if it wouldn't be better to pull up the last two methods into `EntityPersister` if needed and deprecate the whole interface.

Reply:
I'm fine with this, so I made the change.

I also introduced a new version of `IdentityColumnSupportImpl#buildGetGeneratedKeysDelegate` which uses `EntityPersister` and deprecates the old one.